src/main/java/bridge/controller/BridgeController.java

```diff
@@ -0,0 +1,65 @@
+package bridge.controller;
+
+import bridge.BridgeGame;
+import bridge.domain.BridgeSize;
+import bridge.domain.Direction;
+import bridge.domain.GameCommand;
+import bridge.domain.MovingStatus;
+import bridge.view.InputView;
+import bridge.view.OutputView;
+import java.util.List;
+import java.util.function.Supplier;
+
+public class BridgeController {
+
+    private final BridgeGame bridgeGame;
+
+    public BridgeController(BridgeGame bridgeGame) {
+        this.bridgeGame = bridgeGame;
+    }
+
+    public void run() {
+        OutputView.printStartMessage();
+
+        BridgeSize bridgeSize = repeatReadForInvalid(this::getBridgeSize);
+
+        List<String> bridge = bridgeGame.makeBridge(bridgeSize.getSize());
+
+        while (bridgeGame.canMove(bridge)) {
+            Direction movingDirection = repeatReadForInvalid(this::getDirection);
+            MovingStatus movingStatus = bridgeGame.move(bridge, movingDirection);
+            OutputView.printMap(movingStatus);
+            if (movingStatus.cannotCross()) {
+                GameCommand gameCommand = repeatReadForInvalid(this::getGameCommand);
+                if (gameCommand.isRestart()) {
+                    bridgeGame.retry();
+                    continue;
+                }
+                break;
+            }
```

Comment:
인덴트가 깊어졌는데요, 임시방편이 되겠지만, 이번의 경우는 조건식을 다르게 설정해 주면 인덴트를 줄일 수 있을 것 같습니다.
```suggestion
            if (! movingStatus.cannotCross()) {
                continue;
            }
            GameCommand gameCommand = repeatReadForInvalid(this::getGameCommand);
            if (gameCommand.isRestart()) {
                bridgeGame.retry();
                continue;
            }
            break;
```

Reply:
끝나갈 때 해당 로직이 생각나서 급하게 작성하다보니 인덴트가 깊어지는 문제를 피할 수 없었습니다ㅠㅠ
깔끔한 에시 코드 감사합니다!!